### ISSUE TITLE
mosaic focus crash

### ISSUE THREAD

**Issue by dstndstn** [legacysurvey/obsbot#32]
tonight mosaic_focus.py caused a copilot crash on our focus frame.  Running manually

python mosaic_focus.py /mosaic3/data3/observer/20160510/mos3.88174.fits

Gives the error message:

mosaic_focus.py:203: RuntimeWarning: divide by zero encountered in divide
  wt = 1. / Ysig

**Comment by dstndstn** [comment]
This came from too few stars matching PS1 and then resulting failure.
Partially addressed by 441534d

**Comment by weaverba137** [comment]
It looks like we experienced this exact same issue on 2017-02-05.  There are some additional aspects to this problem:

1. When copilot is running in the mayall-idl VNC, and it crashes, it takes down the xterm it is running in with it.  So you can tell that it crashed (if you're fast enough to notice), but can't actually see the error message, because the xterm vanishes.
2. After copilot crashes, it doesn't seem possible to create a new xterm in the VNC, even when clicking on xterm in the Applications menu.  The only way to restart copilot is to destroy the VNC and restart it.

**Comment by dstndstn** [comment]
```
$ python mosaic_focus.py \$MOS3_DATA/mos3107166.fits
...
NO SOURCES DETECTED
Traceback (most recent call last):
  File "mosaic_focus.py", line 461, in <module>
    meas.run(ps=ps, plotfn=opt.plotfn)
  File "mosaic_focus.py", line 94, in run
    if meas['nmatched'] < 10:
KeyError: 'nmatched'
```

Same for 3107167.

3107168 detects only 3 stars and then bails out cleanly, ditto 3107170.

3107174 succeeds.

**Comment by dstndstn** [comment]
To address the problem of the copilot window closing when the copilot crashes, I changed the ~/bin/xt_copilot script, adding "-hold" to the xterm command.  This should keep the window open when copilot crashes.

**Comment by dstndstn** [comment]
Also, I added a level of indirection: instead of running "python copilot.py" directly, the ~/bin/xt_copilot script now runs the ~/bin/run_copilot script, which prints "COPILOT CRASHED!" after the copilot.py command finishes.

**Comment by dstndstn** [comment]
I confirm that I cannot start a new xterm from the window manager menu.  This is not related to the copilot per se, but is obviously something we should fix...

**Comment by dstndstn** [comment]
I made the ~/.vnc/xstartup script start an extra xterm, just to make it easier to restart the copilot window by typing, eg, "~/bin/xt_copilot".

The actual copilot focus crash is fixed in 4fef23c
